pkg/sql/logictest/testdata/logic_test/mixed_version_can_login

```diff
@@ -21,7 +21,7 @@ SELECT crdb_internal.create_tenant(1001)
 upgrade 1
 
 query B
-SELECT crdb_internal.node_executable_version() SIMILAR TO '22.2-%'
+SELECT crdb_internal.node_executable_version() SIMILAR TO '23.1-%'
```

Comment:
I checked the output and it returns `23.1`, so there is no match to `23.1-%`. We can probably change this (and others) to 
```
SELECT crdb_internal.node_executable_version() = '23.1'
```

Reply:
aah, yes - that makes sense, thanks for catching :) - fixed!